dev_guide/app_tutorials/ruby_on_rails.adoc

```diff
@@ -0,0 +1,344 @@
+
+----
+$ oc new-project rails-app --description="My Rails application" --display-name="Rails Application"
+----
+
+After creating the the `rails-app` link:../../dev_guide/projects.html[project], you will be automatically switched to the new project.
+
+Deploying your application in OpenShift will be done in three steps:
+
+* Creating database link:../../architecture/core_concepts/pods_and_services.html#services[service] from OpenShift link:../../using_images/db_images/postgresql.html[PostgreSQL image]
+* Creating frontend link:../../architecture/core_concepts/pods_and_services.html#services[service] from OpenShift link:../../using_images/sti_images/ruby.html[Ruby 2.0 builder image] and your Ruby on Rails source code, which we wire with the database service
+* Creating a route for your application.
+
+=== Creating database service
+
+Our Rails application will expect a running database link:../../architecture/core_concepts/pods_and_services.html#services[service] running. For this service we will use link:http://www.postgresql.org/[PostgeSQL] database link:../../using_images/db_images/postgresql.html[image].
+
+To create the database link:../../architecture/core_concepts/pods_and_services.html#services[service] you will use the link:../../dev_guide/new_app.html[oc new-app] command. To this command you will need to pass some necessary link:../../using_images/db_images/postgresql.html#environment-variables[environment variables] which will be used inside the database container. These link:../../db_images/postgresql.html#environment-variables[environment variables] are required to set the username, password, and name of the database. You can change the values of these link:../../db_images/postgresql.html#environment-variables[environment variables] to anything you would like. The variables we are going to be setting are as follows:
+
+* POSTGRESQL_DATABASE
+* POSTGRESQL_USER
+* POSTGRESQL_PASSWORD
+
+By setting these variables when creating the PostgreSQL database, the image will ensure that:
+
+* A database exists with the specified name
+* A user exists with the specified name
+* The user can access the specified database with the specified password
+
+To do that type following command with your desired values:
+
+----
+$ oc new-app postgresql -e POSTGRESQL_DATABASE=db_name -e POSTGRESQL_USER=username -e POSTGRESQL_PASSWORD=password
+----
+
+In case you would like to also set password for the database administrator, append the previous command with:
+----
+-e POSTGRESQL_ADMIN_PASSWORD=admin_pw
+----
+
+To watch the progress of this command type:
+
+----
+$ oc get pods --watch
+----
+
+=== Creating frontend service
+
+[WARNING]
+====
+*ToBeDone*:
+We in OpenShift need a way how specify deployment hooks upon the deployment process. The use case in the application is the set the databse migration(and maybe also the DB seeding) upon the deployment. For now the hooks can be specified only in the template.
```

Comment:
hups, when I was shuffling with the structure of the docs I've added the docs for the `hooks` into the old docs in the old `app_architecture` and when I was nuking the section I'm deleted them... 
like a boss :+1: